internal/resource/http.go

```diff
@@ -212,10 +213,10 @@ func (c HttpClient) getReaderWithHeader(ctx context.Context, url string, header
 		}
 	}
 
+	ctx, cancelFn := context.WithCancel(context.Background())
 	if c.timeout != 0 {
-		ctxTo, cancel := context.WithTimeout(ctx, c.timeout)
-		ctx = ctxTo
-		defer cancel()
+		cancelFn()
+		ctx, cancelFn = context.WithTimeout(context.Background(), c.timeout)
```

Comment:
There's a chance we'll want to create a child context here with a timeout that the request will be tied to. The timeout length logic is specific to the http client (uses a private variable on the http client) so it can't happen on the fetcher side, but the request lifespan outlives this function, so imo we need to create a context here and end it in the fetcher.